src/sage/matroids/constructor.py

```diff
@@ -734,6 +778,10 @@ def Matroid(groundset=None, data=None, **kwds):
             key = 'graph'
         elif is_Matrix(data):
             key = 'matrix'
+        elif isinstance(data, sage.modules.with_basis.morphism.ModuleMorphism) or (
```

Comment:
instead of using `isinstance` with this class, I think it's better to use `isinstance(data, Morphism) and data.category_for() in Modules().WithBasis()` or something like that